docs/src/main/asciidoc/getting-started-testing.adoc

```diff
@@ -1506,3 +1506,94 @@ For `@QuarkusIntegrationTest` tests that result in launcher the application as a
 This can be used by `QuarkusTestResourceLifecycleManager` that need to launch additional containers that the application will communicate with.
 ====
 
+== Testing Components
+
+In Quarkus, the component model is built on top CDI.
+Therefore, Quarkus provides the `QuarkusComponentTest`, a JUnit extension to ease the testing of components and mocking of their dependencies.
+This extension is available in the `quarkus-junit5-component` dependency.
+
+Let's have a component `Foo`:
+
+[source, java]
+----
+package org.acme;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped <1>
+public class Foo {
+
+    @Inject
+    Charlie charlie; <2>
+
+    public String ping() {
+        return charlie.ping();
+    }
+}
+----
+<1> `Foo` is an `@ApplicationScoped` CDI bean.
+<2> `Foo` depends on `Charlie` which declares a method `ping()`.
+
+Then a component test could look like:
+
+[source, java]
+----
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+import org.acme.Charlie;
+import org.acme.Foo;
+
+public class FooTest {
+
+    @RegisterExtension <1>
+    static final QuarkusComponentTest test = new QuarkusComponentTest(Foo.class);
+
+    @Inject
+    Foo foo; <2>
+
+    // Inject a mock that is created automatically...
+    @Inject
+    Charlie charlie; <3>
+
+    @Test
+    public void testPing() {
+        Mockito.when(charlie.ping()).thenReturn("OK"); <4>
+        assertEquals("OK", foo.ping());
+    }
+}
+----
+<1> The `QuarkusComponentTest` extension is configured in a static field.
+<2> The test injects the component under the test.
+<3> The test also injects `Charlie`, a dependency for which a synthetic `@Singleton` bean is registered automatically. The injected reference is an "unconfigured" Mockito mock. 
+<4> We can leverage the Mockito API in a test method to configure the behavior.
+
+=== Lifecycle
```

Comment:
We should probably it very clear that each test is independent of the next one.

Reply:
In fact, it depends on the test instance lifecycle, i.e. if the `@org.junit.jupiter.api.TestInstance` annotation is used or not. By default, a new test instance is created for each test method. In this case, it's really independent. But if `Lifecycle.PER_CLASS` is used then the container instance is shared across all test methods...

Reply:
All the more reason to improve the documentation :)

Reply:
Well, the documentation intentionally sticks with the JUnit definitions (test instance, test method, `before all` test phase, etc.) and leaves it to JUnit to explain the details but I get your point...